python/cuvs/cuvs/neighbors/brute_force/brute_force.pyx

```diff
@@ -0,0 +1,216 @@
+    -------
+    index: cuvs.neighbors.brute_force.FlatIndex
+
+    Examples
+    --------
+
+    >>> import cupy as cp
+    >>> from cuvs.neighbors import brute_force
+    >>> n_samples = 50000
+    >>> n_features = 50
+    >>> n_queries = 1000
+    >>> k = 10
+    >>> dataset = cp.random.random_sample((n_samples, n_features),
+    ...                                   dtype=cp.float32)
+    >>> index = brute_force.build(dataset, metric="cosine")
+    >>> distances, neighbors = brute_force.search(index, dataset, k)
+    >>> distances = cp.asarray(distances)
+    >>> neighbors = cp.asarray(neighbors)
+    """
+
+    dataset_ai = wrap_array(dataset)
+    _check_input_array(dataset_ai, [np.dtype('float32')])
+
+    cdef cuvsResources_t res = <cuvsResources_t>resources.get_c_obj()
+
+    cdef DistanceType c_metric = <DistanceType>DISTANCE_TYPES[metric]
+    cdef FlatIndex idx = FlatIndex()
+    cdef cydlpack.DLManagedTensor* dataset_dlpack = \
+        cydlpack.dlpack_c(dataset_ai)
+
+    with cuda_interruptible():
+        check_cuvs(cuvsBruteForceBuild(
+            res,
+            dataset_dlpack,
+            c_metric,
+            <float>metric_arg,
+            idx.index
+        ))
+        idx.trained = True
+
+    return idx
+
+
+@auto_sync_resources
+@auto_convert_output
+def search(FlatIndex index,
+           queries,
+           k,
+           neighbors=None,
+           distances=None,
+           resources=None):
+    """
+    Find the k nearest neighbors for each query.
+
+    Parameters
+    ----------
+    index : FlatIndex
+        Trained Brute Force index.
+    queries : CUDA array interface compliant matrix shape (n_samples, dim)
+        Supported dtype [float, int8, uint8]
+    k : int
+        The number of neighbors.
+    neighbors : Optional CUDA array interface compliant matrix shape
+                (n_queries, k), dtype int64_t. If supplied, neighbor
+                indices will be written here in-place. (default None)
+    distances : Optional CUDA array interface compliant matrix shape
+                (n_queries, k) If supplied, the distances to the
+                neighbors will be written here in-place. (default None)
+    {resources_docstring}
+
+    Examples
+    --------
+    >>> import cupy as cp
+    >>> from cuvs.neighbors import brute_force
+    >>> n_samples = 50000
+    >>> n_features = 50
+    >>> n_queries = 1000
+    >>> dataset = cp.random.random_sample((n_samples, n_features),
+    ...                                   dtype=cp.float32)
+    >>> # Build index
+    >>> index = brute_force.build(dataset, metric="sqeuclidean")
+    >>> # Search using the built index
+    >>> queries = cp.random.random_sample((n_queries, n_features),
+    ...                                   dtype=cp.float32)
+    >>> k = 10
+    >>> # Using a pooling allocator reduces overhead of temporary array
+    >>> # creation during search. This is useful if multiple searches
+    >>> # are performad with same query size.
```

Comment:
I know that I am one of the people that left this comment in the examples :), but was thinking it'd be neat if we add it instead to the docstring of relevant methods in a "widget" style, similar to what networkx does with cugraph algos, see the bottom of https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.centrality.betweenness_centrality.html

Reply:
This is a great idea. Do one of you mind copy/pasting this comment into a Github issue just so we can keep it on our radar?